RUFAS/output_manager.py

```
@@ -410,6 +411,19 @@ def _generate_file_name(self, base_name: str, extension: str = "json") -> str:
         timestamp: str = self._get_timestamp(include_millis=False)
         return f"{base_name}_{timestamp}.{extension}"
 
+    def _generate_graph_path(self, save_path: str, graph_info: dict, filter_file: str) -> str:
+        graph_directory = os.path.join(save_path, "graphics", "om")
+        try:
+            Path(graph_directory).mkdir(parents=True, exist_ok=True)
+        except Exception as e:
+            raise e
```

At this point, we have run the simulation and have the results ready. If we can't make the directory, we still can save the the result as JSON or CSV. Raising the exception here halts the execution. Instead, let's report the error to OM.errors_pool and move on (use self.add_error() )

---

If I move this function `generate_graph_path` into the GraphGenerator class, then I cannot add do `om.add_error()`. This would be a circular import. (OM is importing graph_generator.py).

---

I guess I can extract the highlighted part (the part for making the directory) and move it into `utils.py` as a util function. How does this sound? @PooyaHekmati